ldpd/ldpd.c

```diff
@@ -1349,6 +1349,12 @@ merge_af(int af, struct ldpd_af_conf *af_conf, struct ldpd_af_conf *xa)
 	if ((af_conf->flags & F_LDPD_AF_EXPNULL) !=
 	    (xa->flags & F_LDPD_AF_EXPNULL))
 		change_egress_label = 1;
+
+	/* changing config of host only fec filtering */
+	if ((af_conf->flags & F_LDPD_AF_ALLOCHOSTONLY)
+	    != (xa->flags & F_LDPD_AF_ALLOCHOSTONLY))
+		change_egress_label = 1;
```

Comment:
Reusing `lde_change_egress_label()` would be insufficient here for two reasons:
* It only withdraws previously advertised null labels to all peers (using wildcard FECs);
* It skips FECs whose local label is null (either exp-null or imp-null) when reevaluating all local labels.

When "[no] label local allocate" is issued (either with the "host-routes" or "for ACL" options), ideally we should try to do the following:
* For each FEC, reevaluate it's local label;
* If the local label has changed and it's different from `NO_LABEL`, send a label mapping to all peers advertising the new label (there's no need to send an explicit label withdraw before).
* If the local label has changed to `NO_LABEL`, send a label withdraw to all peers.
* If the local label hasn't changed, do nothing.

Another option would be to send a wildcard label withdraw to all peers then send label mappings for each FEC as appropriate. That would be slightly easier to implement, but unnecessarily disruptive.

Reply:
Thanks Renato,   I will add new routine for this.